tools/tpm2_createpolicy.c

```diff
@@ -320,27 +160,23 @@ int execute_tool(int argc, char *argv[], char *envp[], common_opts_t *opts,
     (void) opts;
     (void) envp;
 
-    createpolicypcr_ctx ctx = { 
-        .sapi_context = sapi_context, 
-        .policy_session = NULL,
-        .policy_digest = TPM2B_TYPE_INIT(TPM2B_DIGEST, buffer),
-        .pcr_index = 0,
-        .max_supported_pcrs = 0,
-        .hash_alg = TPM_ALG_SHA1,
-        .policyfile = NULL,
-        .raw_pcr_file = NULL,
-        .pcr_flags.policy_type_pcr_flag = false,
-        .pcr_flags.pcr_index_flag = false,
-        .pcr_flags.raw_pcr_flag = false
+    CREATE_POLICY_CTX pctx = {
```

Comment:
tool context structures should be private to the tool. You're exposing a an internal policy building API. Which I am assuming more than just tpm2_createpolicy() is going to use.

You need to seperate what should be a part of the polict structure, and the rest of the arguments. I don't see sapi_context applying to create_policy structure.

I think a better name would be to call it a tpm2_policy structure, and pass sapi_context as an argument to the tpm2_policy_build() routines.